app/javascript/dashboard/routes/dashboard/settings/inbox/components/WeeklyAvailability.vue

```diff
@@ -0,0 +1,188 @@
+import SettingsSection from 'dashboard/components/SettingsSection';
+import BusinessDay from './BusinessDay';
+import {
+  timeSlotParse,
+  timeSlotTransform,
+  defaultTimeSlot,
+  timeZoneOptions,
+} from '../helpers/businessHour';
+
+const DEFAULT_TIMEZONE = 'America/Los_Angeles';
+
+export default {
+  components: {
+    SettingsSection,
+    BusinessDay,
+  },
+  mixins: [alertMixin],
+  props: {
+    inbox: {
+      type: Object,
+      default: () => ({}),
+    },
+  },
+  data() {
+    return {
+      isBusinessHoursEnabled: false,
+      unavailableMessage: this.$t(
+        'INBOX_MGMT.BUSINESS_HOURS.UNAVAILABLE_MESSAGE_DEFAULT'
+      ),
+      timeZone: DEFAULT_TIMEZONE,
+      dayNames: {
+        0: 'Sunday',
+        1: 'Monday',
+        2: 'Tuesday',
+        3: 'Wednesday',
+        4: 'Thursday',
+        5: 'Friday',
+        6: 'Saturday',
+      },
+      timeSlots: [...defaultTimeSlot],
+    };
+  },
+  computed: {
+    ...mapGetters({ uiFlags: 'inboxes/getUIFlags' }),
+    hasError() {
+      if (!this.isBusinessHoursEnabled) return false;
+      return this.timeSlots.filter(slot => slot.from && !slot.valid).length > 0;
+    },
+    timeZones() {
+      return [...timeZoneOptions()];
+    },
+  },
+  watch: {
+    inbox() {
+      this.setDefaults();
+    },
+  },
+  mounted() {
+    this.setDefaults();
+  },
+  methods: {
+    setDefaults() {
+      const {
+        working_hours_enabled: isEnabled = false,
+        out_of_office_message: unavailableMessage = this.$t(
+          'INBOX_MGMT.BUSINESS_HOURS.UNAVAILABLE_MESSAGE_DEFAULT'
+        ),
+        working_hours: timeSlots = [],
+        timezone: timeZone,
+      } = this.inbox;
+      const slots = timeSlotParse(timeSlots).length
+        ? timeSlotParse(timeSlots)
+        : defaultTimeSlot;
+      this.isBusinessHoursEnabled = isEnabled;
+      this.unavailableMessage = unavailableMessage;
+      this.timeSlots = slots;
+      this.timeZone =
+        this.timeZones.find(item => timeZone === item.value) ||
+        DEFAULT_TIMEZONE;
+    },
+    onSlotUpdate(slotIndex, slotData) {
+      this.timeSlots = this.timeSlots.map(item =>
+        item.day === slotIndex ? slotData : item
+      );
+    },
+    async updateInbox() {
+      try {
+        const payload = {
+          id: this.inbox.id,
+          formData: false,
+
+          working_hours_enabled: this.isBusinessHoursEnabled,
+          out_of_office_message: this.unavailableMessage,
+          working_hours: timeSlotTransform(this.timeSlots),
+          timezone: this.timeZone.value,
+          channel: {},
+        };
+        await this.$store.dispatch('inboxes/updateInbox', payload);
+        this.showAlert(this.$t('INBOX_MGMT.EDIT.API.SUCCESS_MESSAGE'));
+      } catch (error) {
+        this.showAlert(this.$t('INBOX_MGMT.EDIT.API.SUCCESS_MESSAGE'));
+      }
+    },
+  },
+};
+</script>
+<style lang="scss" scoped>
+.timezone-input-wrap {
+  max-width: 60rem;
+
+  &::v-deep .multiselect {
```

Comment:
I would suggest using another selector, if anything changes in vue upgrade, this would break.

Reply:
IMHO Vue3 upgrade is far away.